Instagram/View Controllers/ProfileViewController.m

```diff
@@ -5,16 +5,122 @@
 //  Created by Megan Miller on 6/27/22.
 //
 
+#import <Parse/Parse.h>
 #import "ProfileViewController.h"
+#import "ProfileCell.h"
 
-@interface ProfileViewController ()
+@interface ProfileViewController () <UICollectionViewDelegate, UICollectionViewDataSource, UIImagePickerControllerDelegate, PFSubclassing>
+
+@property (weak, nonatomic) IBOutlet UIImageView *profileImageView;
+@property (weak, nonatomic) IBOutlet UIButton *editProfileButton;
+@property (weak, nonatomic) IBOutlet UICollectionView *collectionView;
+@property (weak, nonatomic) IBOutlet UILabel *usernameLabel;
+
+@property (nonatomic, strong) NSMutableArray *userPosts;
 
 @end
 
 @implementation ProfileViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    
+    self.collectionView.delegate = self;
+    self.collectionView.dataSource = self;
+    
+    // TODO: async?
+    [self.user fetchIfNeeded];
+    // TODO: if statement
+    if (!(self.user) || [self.user isEqual:[PFUser currentUser]]) {
+        self.user = [PFUser currentUser];
+        [self.editProfileButton setHidden:NO];
+        [self fetchUserPosts];
+    } else {
+        [self.editProfileButton setHidden:YES];
+        [self fetchUserPosts];
+    }
+    
+    PFFileObject *userImageData = [self.user valueForKey:@"profilePicture"];
+    [userImageData getDataInBackgroundWithBlock:^(NSData * _Nullable data, NSError * _Nullable error) {
+        if (data) {
+            UIImage *profileImage = [UIImage imageWithData:data];
+            [self.profileImageView setImage:profileImage];
+        }
+        else {
+            NSLog(@"Unable to load image.");
+        }
+    }];
+    
+    [self.usernameLabel setText:self.user.username];
+}
+
+- (void)fetchUserPosts {
+    PFQuery *query = [PFQuery queryWithClassName:@"Post"];
+    [query whereKey:@"author" equalTo:self.user]; // TODO: compares pointers?
+    [query orderByDescending:@"createdAt"];
+    query.limit = 20;
+
+    [query findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
+        if (posts != nil) {
+            self.userPosts = (NSMutableArray *)posts;
+            [self.collectionView reloadData];
+        }
+    }];
+}
+
+- (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return self.userPosts.count;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    ProfileCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"ProfileCell" forIndexPath:indexPath];
```

Comment:
Protip: If you need the name of the class in string form, you can use `NSStringFromClass([ProfileCell class])`